### PR TITLE
Fix prayer to next combat level calculation

### LINKED ISSUE #7103: Calculate next combat level wrong
While hovering over my cmb lvl, it said I needed 3 more prayer level to advance a cmb lvl while actually only needing 1 prayer level.

All cmb stats at 99.
Prayer from 89 to 90 calculates wrong.

### PR DESCRIPTION
**Pull request by Nightfirecat**
Fixes #7103

### CODE REVIEW

File: runelite-client/src/test/java/net/runelite/client/plugins/combatlevel/CombatLevelPluginTest.java [hunk 1]
```diff
@@ -262,8 +253,27 @@ public void testPlayerZezima()
...
-			player.getCombatLevel() + 1, PRAY_MULT);
-		assertEquals(4, correctPrayer(client.getRealSkillLevel(Skill.PRAYER), prayNeed));
+		assertEquals(4, calcLevelsPray(baseValues.get("base") + baseValues.get("max"),
+			player.getCombatLevel() + 1, client.getRealSkillLevel(Skill.PRAYER)));
+	}
+
+	@Test
+	public void testPrayerLevelsNeeded()
+	{
+		// Check an edge case where the needed prayer levels to next combat could be calculated to be 3 instead of 1
```
Comment (Adam-) [inline runelite-client/src/test/java/net/runelite/client/plugins/combatlevel/CombatLevelPluginTest.java]: How is this an "edge case"?

Comment (Nightfirecat) [reply runelite-client/src/test/java/net/runelite/client/plugins/combatlevel/CombatLevelPluginTest.java]: Good point, I'll move this to the commit message.
